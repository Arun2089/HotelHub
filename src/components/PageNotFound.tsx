import ErrorPage from "./ErrorPage";
import pageImg from "@assets/404errorimg.png";

const PageNotFound = () => {
  return (
    <ErrorPage
      image={pageImg}
      customRender={
        <div className="mt-8">
          <p className="max-w-3xl pb-4 text-sm lg:text-xl text-center text-primary-main">
            <h1 className="text-2xl lg:text-4xl font-semibold text-primary-light">
              404 Page Not Found
            </h1>
            Sorry, the page you're looking for isn't here, Please check the URL
            or return to the Dashboard Page.
          </p>
        </div>
      }
    />
  );
};

export default PageNotFound;
