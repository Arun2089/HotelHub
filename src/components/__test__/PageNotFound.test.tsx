import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import PageNotFound from "../PageNotFound";

describe("Forbidden component", () => {
  it("snapshot test", () => {
    const { asFragment } = render(<PageNotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    render(<PageNotFound />);
    const image = screen.getByRole("img");
    const backLink = screen.getByText("Back");
    const HomeBtn = screen.getByText("Home");

    expect(image).toBeInTheDocument();
    expect(backLink).toBeInTheDocument();
    expect(HomeBtn).toBeInTheDocument();
  });
});
