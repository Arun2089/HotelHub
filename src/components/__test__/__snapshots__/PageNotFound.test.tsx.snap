// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forbidden component snapshot test 1`] = `
<DocumentFragment>
  <div
    class="h-screen w-full flex items-center justify-center lg:justify-start flex-col bg-white MuiBox-root css-0"
  >
    <img
      class="w-full md:w-1/2 h-auto  lg:h-max-[50vh] MuiBox-root css-0"
      src="404errorimg.png"
    />
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge mr-8 css-9y1egq-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="HomeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </span>
        Home
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
